lib/src/handler.dart

```diff
@@ -83,15 +83,13 @@ class ShelfTestHandler {
         throw TestFailure(message);
       }
 
-      var response = await expectation.handler(request);
-      if (response != null) return response;
-
-      throw TestFailure('$description handler returned null for $requestInfo.');
+      return await expectation.handler(request);
     } on HijackException catch (_) {
       rethrow;
     } catch (error, stackTrace) {
       _zone.handleUncaughtError(error, stackTrace);
-      return null;
+      // We can't return null here, the handler type doesn't allow it.
+      return Response.internalServerError(body: 'Response Hijacked');
```

Comment:
I was a bit unsure what to do here, we are not allowed to return null.

Reply:
I don't think `'Response Hijacked'` would be the appropriate text if this is the pattern we use.

What would happen if we also `rethrow` here?

Reply:
I really have no idea what the intention of this code is. I don't see any existing tests for the hijack behavior so I can't tell what is expected...

Reply:
Oh wait I just realized that this wasn't even in the hijack block lol, so this is just catching general errors.

If I rethrow it makes the tests fail. Maybe just a `Response.notFound('')` would be appropriate?

Reply:
Ya this is intentionally forwarding the errors to the original zone where the ShelfTestHandler was created. Rethrowing makes it show up in the current zone as well.

I think just some sort of placeholder response is the correct thing here, even though its a bit weird.

Reply:
I went with `return Response.internalServerError(body: '$error');` as that does kind of make sense? I don't think this is super important actually.

Reply:
`return Response.internalServerError(body: '$error');` SGTM.